src/EnergyPlus/api/plugin.py

```diff
@@ -49,53 +47,53 @@ def _detect_overridden(self) -> List[str]:
                 diff.remove(known_skip)
         return diff
 
-    def on_begin_new_environment(self) -> int:
+    def on_begin_new_environment(self, state) -> int:
```

Comment:
Added state as an argument to the plugin functions...all user-overridden functions will need to accept this argument, and then they can use it to call back into the API.